walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,174 @@
+
+    for (TimeRange emptySlot : timeTable) {
+      if (emptySlot.duration() >= duration) {
+        suitableSlots.add(emptySlot);
+      }
+    }
+
+    return suitableSlots;
+  }
+
+  private TimeTablePair registerRelevantEvents(final Collection<Event> events,
+                                               final Collection<String> mandatoryAttendees,
+                                               final Collection<String> optionalAttendees) {
+    /**
+     * The timeTable collection holds TimeRanges which represent empty time slots for
+     * all the people invited
+     */
+    List<TimeRange> timeTable = new ArrayList<>();
+    List<TimeRange> optionalTimeTable = new ArrayList<>();
+    timeTable.add(TimeRange.WHOLE_DAY);
+    optionalTimeTable.add(TimeRange.WHOLE_DAY);
+
+    for (Event event : events) {
+      if (relevantEvent(event, mandatoryAttendees)) {
+        splitTimeTable(timeTable, event.getWhen());
+        splitTimeTable(optionalTimeTable, event.getWhen());
+      }
+
+      if (relevantEvent(event, optionalAttendees)) {
+        splitTimeTable(optionalTimeTable, event.getWhen());
+      }
+    }
+
+    return new TimeTablePair(timeTable, optionalTimeTable);
+  }
+
+  private Boolean relevantEvent(final Event event, final Collection<String> relevantPeople) {
+    for (String person : relevantPeople) {
+      if (event.getAttendees().contains(person)) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  private void splitTimeTable(final List<TimeRange> originalTimeTable,
+                              final TimeRange timeSplitter) {
+    /**
+     * Possible cases:
+     * 1. originalTimeTable:  [--------------------------]
+     *    splitter:                     [-----]
+     *    result:             [---------]     [----------]
+     *
+     * 2. originalTimeTable:  [------]   ...   [---------]
+     *    splitter:                [-------------]
+     *    result:             [----]             [-------]
+     *
+     * 3. originalTimeTable:  [------]   ...  [----------]
+     *    splitter:                     [-----]
+     *    result:             [------]        [----------]
+     *
+     * 4. originalTimeTable:  [------]           [-------]
+     *    splitter:                     [-----]
+     *    result:             [------]           [-------]
+     *
+     * Mention: for this method, it will only matter if the splitter is contained or not
+     */
+    if (originalTimeTable.isEmpty()) {
+      return;
+    }
+    int firstRelevantTimeslotIdx = TimeRange.lowerBound(originalTimeTable, timeSplitter);
+
+    if (originalTimeTable.get(firstRelevantTimeslotIdx).contains(timeSplitter)) {  // Case 1
+      splitAtTimeslot(originalTimeTable, timeSplitter, firstRelevantTimeslotIdx);
+    } else {  // Case 2, 3, 4
+      modifyOverlappingSlots(originalTimeTable, timeSplitter, firstRelevantTimeslotIdx);
+    }
+  }
+
+  private void splitAtTimeslot(final List<TimeRange> timeTable, final TimeRange splitter,
```

Comment:
A short comment would help a reader understand why exactly you are using > and < (possibility of same ending time and the slot completely disappearing). Since this hasn't been mentioned in 1.
It is also good to add side effects, like the modification of the timeTable, in comment / documentation.